src/openloco/ui.cpp

```diff
@@ -780,87 +781,99 @@ namespace openloco::ui
         SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_BUTTON_RETURNKEY_DEFAULT, title.c_str(), message.c_str(), window);
     }
 
+    config::resolution_t getDisplayResolutionByMode(config::screen_mode mode, int32_t displayIndex)
+    {
+        config::resolution_t outputResolution = { 0, 0 };
+        auto& config = config::get_new();
+        switch (mode)
+        {
+        case config::screen_mode::window:
+            if (config.display.window_resolution.width > 0 && config.display.window_resolution.height > 0)
+                outputResolution = { config.display.window_resolution.width, config.display.window_resolution.height };
```

Comment:
You're still duplicating the data. `config.display.window_resolution` is already a `resolution_t` — you can just assign it directly: `outputResolution = config.display.window_resolution;`.

Preferably, you'd change `config::resolution_t outputResolution` to `config::resolution_t* outputResolution` though (default nulltptr). This would prevent copying the object (or creating a new one entirely). You'd then assign it as: `outputResolution = &config.display.window_resolution;`

Reply:
I've read a bunch about what the & and * symbols mean, and I think I understand now.

If I declare it with `config::resolution_t* outputResolution` I can set the `outputResolution` variable by setting the pointer to the existing address at `&config.display.window_resolution` or `&config.fullscreen_resolution.window_resolution`

However I am unable to set a default value manually like:
`outputResolution = { 640, 480 };`

Would this be acceptable to set a default value when there is none in the config, or am I missing something?
```
config::resolution_t defaultResolution = { 640, 480 };
outputResolution = &defaultResolution;
```

Reply:
Yes, that seems like an appropriate solution. If you don't want anything to change the default resolution, you should mark it `const`, too. Aside from being good practice, this will enable the compiler to perform additional optimisations, too.